### DISCUSSION TITLE
Inconsistent endpoint paths generation

### DISCUSSION THREAD

**Discussion by ecabrerar** [smallrye/smallrye-open-api#1085] [General]
Hi all...

We notice sometimes openapi generated the endpoint paths like we are expecting, but sometimes generate the path without the  JAX-RS Application prefix.

For example:
Our JAX-RS services use the prefix `servicios`, we are expecting paths like '/servicios/xxxx/xxx'

/servicios/aplicacion/version  (when is ok)
/aplicacion/version  (when is wrong)

Application Server: WildFly Full 24.0.1.Final 
Microprofile:
microprofile-openapi-api : 3.0
microprofile-config-api: 3.0.1

[openapi-failed.txt](https://github.com/smallrye/smallrye-open-api/files/8198818/openapi-failed.txt)
[openapi-ok.txt](https://github.com/smallrye/smallrye-open-api/files/8198819/openapi-ok.txt)

**Comment by MikeEdgar** [comment]
@ecabrerar can you share a simple reproducer app?

**Reply by ecabrerar** [reply]
I could not reproduce it with a single maven project. I'm using openapi with a multi-module maven project. I did try to structure a reproducer app with multiple maven modules like the real project, but it was not posible to reproduce it.

After removing some old Java EE dependencies of the project, we have some improve, the paths are generated like we are expecting most of the time. I'll let you know if I can reproduce the same situation with a reproducer app or find a solution.

Thanks for your quick response!.

**Reply by MikeEdgar** [reply]
@ecabrerar you don't possibly have multiple `Application` classes do you?

**Reply by ecabrerar** [reply]
We don't notice there are two Application class in the core module. 

The first one:
@ApplicationPath("servicios")
public class ApplicationServicesPath extends Application{

}

And the second one:
public final class RestConfiguration extends Application{
@Override
	public Set<Class<?>> getClasses() {
	
	}
}

In the last one, they use a ServiceLoader to load all classes with @Path annotations.

Now, I could reproduce the same behaviour with a reproducer app.

Thank you for your help!.

**Reply by codespearhead** [reply]
@ecabrerar I'm not sure I understand your last message. Does that mean the problem is solved?

If so, can you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so that it doesn't show up [in the discussion tab](https://github.com/smallrye/smallrye-open-api/discussions) as unanswered?
